### PR TITLE
[CLOUD-1173] Add displayName to all template parameters

### PR DESCRIPTION
**Pull request by jwendell**

### CODE REVIEW

File: amq/amq62-basic.json [hunk 1]
```diff
@@ -18,62 +18,72 @@
         {
             "description": "The name for the application.",
             "name": "APPLICATION_NAME",
+            "displayName": "Application Name",
```
Comment (luksa) [inline amq/amq62-basic.json]: Can you move this up to just above "description"? IMHO this would make it a bit cleaner.

Comment (luksa) [inline amq/amq62-basic.json]: Can you move this up to just above "description"? IMHO this would make it a bit cleaner.

File: amq/amq62-basic.json [hunk 2]
```diff
@@ -18,62 +18,72 @@
...
             "description": "List of packages that are allowed to be serialized for use in ObjectMessage, separated by commas. If your app doesn't use ObjectMessages, leave this blank. This is a security enforcement. For the rationale, see http://activemq.apache.org/objectmessage.html",
             "name": "MQ_SERIALIZABLE_PACKAGES",
+            "displayName": "Serializable Packages",
             "value": "",
             "required": false
         },
         {
             "description": "User name for standard broker user. It is required for connecting to the broker. If left empty, it will be generated.",
             "name": "MQ_USERNAME",
+            "displayName": "Username",
```
Comment (luksa) [inline amq/amq62-basic.json]: Will it be clear what the username is for? Shouldn't it be "MQ Username" or something similar. I feel like just "Username" is not specific enough and may be confusing to the user.

File: amq/amq62-persistent.json [hunk 3]
```diff
@@ -18,74 +18,86 @@
...
             "description": "List of packages that are allowed to be serialized for use in ObjectMessage, separated by commas. If your app doesn't use ObjectMessages, leave this blank. This is a security enforcement. For the rationale, see http://activemq.apache.org/objectmessage.html",
             "name": "MQ_SERIALIZABLE_PACKAGES",
+            "displayName": "Serializable Packages",
             "value": "",
             "required": false
         },
         {
             "description": "Size of persistent storage for database volume.",
             "name": "VOLUME_CAPACITY",
+            "displayName": "Volume Capacity",
```
Comment (luksa) [inline amq/amq62-persistent.json]: Maybe "Database Volume Capacity" to make it clear.

**Comment by jwendell** [comment]
just pushed a new version

File: amq/amq62-basic.json [hunk 4]
```diff
@@ -16,62 +16,72 @@
...
         },
         {
+            "displayName": "Serializable Packages",
             "description": "List of packages that are allowed to be serialized for use in ObjectMessage, separated by commas. If your app doesn't use ObjectMessages, leave this blank. This is a security enforcement. For the rationale, see http://activemq.apache.org/objectmessage.html",
             "name": "MQ_SERIALIZABLE_PACKAGES",
             "value": "",
             "required": false
         },
         {
+            "displayName": "Username",
```
Comment (rcernich) [inline amq/amq62-basic.json]: Capitalize

File: amq/amq62-persistent-ssl.json [hunk 5]
```diff
@@ -16,104 +16,121 @@
...
         },
         {
+            "displayName": "Serializable Packages",
             "description": "List of packages that are allowed to be serialized for use in ObjectMessage, separated by commas. If your app doesn't use ObjectMessages, leave this blank. This is a security enforcement. For the rationale, see http://activemq.apache.org/objectmessage.html",
             "name": "MQ_SERIALIZABLE_PACKAGES",
             "value": "",
             "required": false
         },
         {
+            "displayName": "Volume Capacity",
```
Comment (rcernich) [inline amq/amq62-persistent-ssl.json]: I think @luksa had commented about this being Database Volume Capacity

File: amq/amq62-persistent-ssl.json [hunk 6]
```diff
@@ -16,104 +16,121 @@
...
         },
         {
+            "displayName": "Keystore Password",
             "description": "Password for accessing SSL keystore",
             "name": "AMQ_KEYSTORE_PASSWORD",
             "value": "",
             "required": true
         },
         {
+            "displayName": "Storage Limit",
```
Comment (rcernich) [inline amq/amq62-persistent-ssl.json]: I think this should be A-MQ Storage Usage Limit, as it's particular to A-MQ configuration.

File: datagrid/datagrid65-mysql-persistent.json [hunk 7]
```diff
@@ -16,198 +16,232 @@
...
         {
+            "displayName": "Database Password",
             "description": "Database user password",
             "name": "DB_PASSWORD",
             "from": "[a-zA-Z0-9]{8}",
             "generate": "expression",
             "required": true
         },
         {
+            "displayName": "Minimum Pool Size",
```
Comment (rcernich) [inline datagrid/datagrid65-mysql-persistent.json]: These parameters should probably be prefaced with "Datasource" as they apply to the datasource configuration.

File: datagrid/datagrid65-mysql-persistent.json [hunk 8]
```diff
@@ -16,198 +16,232 @@
...
             "required": false
         },
         {
+            "displayName": "MySQL Lower Case Table Names",
             "description": "Sets how the table names are stored and compared.",
             "name": "MYSQL_LOWER_CASE_TABLE_NAMES",
             "required": false
         },
         {
+            "displayName": "Maximum number of connections",
```
Comment (rcernich) [inline datagrid/datagrid65-mysql-persistent.json]: These should be prefaced with MySQL, as they apply to the MySQL DB configuration.

File: datagrid/datagrid65-mysql-persistent.json [hunk 9]
```diff
@@ -16,198 +16,232 @@
...
         {
+            "displayName": "Datavirt Cache Names",
             "description": "Comma-separated list of caches to configure for use by Red Hat JBoss Data Virtualization for materialization of views.  Three caches will be created for each named cache: <name>, <name>_staging and <name>_alias.",
             "name": "DATAVIRT_CACHE_NAMES",
             "value": "",
             "required": false
         },
         {
+            "displayName": "Encryption Requires SSL Client Authentication?",
             "description": "",
```
Comment (rcernich) [inline datagrid/datagrid65-mysql-persistent.json]: We should probably include a description here too.  The name is not very clear as to what this parameter does.

File: datagrid/datagrid65-postgresql-persistent.json [hunk 10]
```diff
@@ -16,183 +16,214 @@
...
             "required": false
         },
         {
+            "displayName": "Transaction Isolation",
             "description": "Sets transaction-isolation for the configured datasource.",
             "name": "DB_TX_ISOLATION",
             "required": false
         },
         {
+            "displayName": "Maximum number of connections",
```
Comment (rcernich) [inline datagrid/datagrid65-postgresql-persistent.json]: These should be prefaced with PostgreSQL, since they apply to the PostgreSQL db config.

File: decisionserver/decisionserver62-amq-s2i.json [hunk 11]
```diff
@@ -16,161 +16,187 @@
...
         },
         {
+            "displayName": "Hostname for the HTTPS route",
             "description": "Custom hostname for https service route.  Leave blank for default hostname, e.g.: secure-<application-name>-<project>.<default-domain-suffix>",
             "name": "HOSTNAME_HTTPS",
             "value": "",
             "required": false
         },
         {
+            "displayName": "Source Code Repository Location",
```
Comment (rcernich) [inline decisionserver/decisionserver62-amq-s2i.json]: Please refer to the displayNames used in the datavirt templates, as we should be using the same names for all like parameters.  Specifically, these are common to all s2i templates.  Also, I haven't checked the other parameters like HTTP_NAME, ssl configuration or jgroups configuration, but those parameters should all match what is used in the datavirt templates (i.e. we should be internally consistent).

File: datagrid/datagrid65-mysql-persistent.json (hunk 9 above)
Comment (jwendell) [inline datagrid/datagrid65-mysql-persistent.json]: OK, will leave that for another commit, once I'm doing these changes in batch

**Comment by jwendell** [comment]
Just force-pushed new version of the patch

File: amq/amq62-persistent-ssl.json [hunk 12]
```diff
@@ -16,104 +16,121 @@
...
         },
         {
+            "displayName": "A-MQ Serializable Packages",
             "description": "List of packages that are allowed to be serialized for use in ObjectMessage, separated by commas. If your app doesn't use ObjectMessages, leave this blank. This is a security enforcement. For the rationale, see http://activemq.apache.org/objectmessage.html",
             "name": "MQ_SERIALIZABLE_PACKAGES",
             "value": "",
             "required": false
         },
         {
+            "displayName": "Database Volume Capacity",
```
Comment (rcernich) [inline amq/amq62-persistent-ssl.json]: This isn't a database.  This is the volume size used by A-MQ

File: amq/amq62-persistent-ssl.json [hunk 13]
```diff
@@ -16,104 +16,121 @@
...
         },
         {
+            "displayName": "Secret Name",
             "description": "Name of a secret containing SSL related files",
             "name": "AMQ_SECRET",
             "value": "amq-app-secret",
             "required": true
         },
         {
+            "displayName": "Trust Store",
```
Comment (rcernich) [inline amq/amq62-persistent-ssl.json]: Maybe Trust Store File Name is more descriptive?

File: amq/amq62-persistent-ssl.json [hunk 14]
```diff
@@ -16,104 +16,121 @@
...
         },
         {
+            "displayName": "Trust Store Password",
             "description": "SSL trust store password",
             "name": "AMQ_TRUSTSTORE_PASSWORD",
             "value": "",
             "required": true
         },
         {
+            "displayName": "Keystore",
```
Comment (rcernich) [inline amq/amq62-persistent-ssl.json]: Like truststore, maybe Keystore File Name

File: datagrid/datagrid65-postgresql-persistent.json [hunk 15]
```diff
@@ -16,183 +16,214 @@
...
         },
         {
+            "displayName": "Database JNDI",
             "description": "Database JNDI name used by application to resolve the datasource, e.g. java:/jboss/datasources/postgresql",
             "name": "DB_JNDI",
             "value": "java:jboss/datasources/postgresql",
             "required": false
         },
         {
+            "displayName": "Database",
```
Comment (rcernich) [inline datagrid/datagrid65-postgresql-persistent.json]: This should probably be Database Name.  Database is a bit ambiguous.  Database Name relates to the catalog or schema within the DB server.

File: datagrid/datagrid65-postgresql.json [hunk 16]
```diff
@@ -16,177 +16,207 @@
...
         },
         {
+            "displayName": "Server Keystore Password",
             "description": "The password for the keystore and certificate",
             "name": "HTTPS_PASSWORD",
             "value": "",
             "required": false
         },
         {
+            "displayName": "Database JNDI",
```
Comment (rcernich) [inline datagrid/datagrid65-postgresql.json]: This should probably be Database JNDI Name

File: decisionserver/decisionserver62-amq-s2i.json [hunk 17]
```diff
@@ -16,161 +16,187 @@
...
         {
+            "displayName": "A-MQ Password",
             "description": "Password for standard broker user. It is required for connecting to the broker. If left empty, it will be generated.",
             "name": "MQ_PASSWORD",
             "from": "[a-zA-Z0-9]{8}",
             "generate": "expression",
             "required": false
         },
         {
+            "displayName": "Mesh Discovery Type",
```
Comment (rcernich) [inline decisionserver/decisionserver62-amq-s2i.json]: For the non amq62-* templates, this should probably be A-MQ Mesh Discovery Type, as it applies to A-MQ and not kieserver, eap, etc.

File: amq/amq62-persistent-ssl.json (hunk 12 above)
Comment (jwendell) [inline amq/amq62-persistent-ssl.json]: Didn't understand. This exactly the same name, description and value as used on every other place. Why should its displayName be different?

File: datagrid/datagrid65-postgresql-persistent.json (hunk 15 above)
Comment (jwendell) [inline datagrid/datagrid65-postgresql-persistent.json]: Yep, I though that too, but the description already says that. Sometimes I did choose to omit some information on the displayName because it's already contained in description. But OK, I'll change that.

File: decisionserver/decisionserver62-amq-s2i.json (hunk 17 above)
Comment (jwendell) [inline decisionserver/decisionserver62-amq-s2i.json]: Can I generalize and change *all* of them (in contrast of only no amq62-*) to include the A-MQ prefix?

File: amq/amq62-persistent-ssl.json (hunk 12 above)
Comment (rcernich) [inline amq/amq62-persistent-ssl.json]: same name, but it doesn't apply to a database.

File: datagrid/datagrid65-postgresql.json (hunk 16 above)
Comment (jwendell) [inline datagrid/datagrid65-postgresql.json]: Ditto

File: datagrid/datagrid65-postgresql-persistent.json (hunk 15 above)
Comment (rcernich) [inline datagrid/datagrid65-postgresql-persistent.json]: Ideally, the displayName should be self-describing.  In this instance, you could argue the description should be enhanced to include something saying that this applies to the catalog, schema, etc. within the database server.

File: decisionserver/decisionserver62-amq-s2i.json (hunk 17 above)
Comment (rcernich) [inline decisionserver/decisionserver62-amq-s2i.json]: That's fine with me.  I hope that you are reviewing all of your changes individually and not simply applying a bulk update and assuming the changes are good for all templates.

File: amq/amq62-persistent-ssl.json (hunk 12 above)
Comment (jwendell) [inline amq/amq62-persistent-ssl.json]: so, 1) does this apply to every amq template? 2) is description wrong too and should be changed? 3) is there any other template/product where the semantic for this variable is other than "database volume capacity"? 4) should we go further and change the variable name to avoid confusions like this in the future?

File: decisionserver/decisionserver62-amq-s2i.json (hunk 17 above)
Comment (jwendell) [inline decisionserver/decisionserver62-amq-s2i.json]: yeah, I'm doing this in a batch mode, referring to variable names as a key in a dictionary. So far so good, except in the cases where the same variable name refers to two things (database volume capacity case). I guess those should be handled separately.

File: amq/amq62-persistent-ssl.json (hunk 12 above)
Comment (rcernich) [inline amq/amq62-persistent-ssl.json]: The description in the amq62-*-persistent templates appears to be incorrect.  For those, this is the size of the mounted volume used for persisting messages.  You can look at the template to see how it's used and at the image scripts to see how that affects the configuration.  A-MQ docs would be the last stop.  If you notice, these amq62-*-persistent templates don't create a database.  When in doubt, look to see how the parameter is used.

Comment (jwendell) [inline amq/amq62-persistent-ssl.json]: Yeah, makes sense. Thanks.

**Comment by jwendell** [comment]
Added a commit to fix the Volume Capacity description issue (and fixed the other issues as usual)

File: amq/amq62-persistent-ssl.json [hunk 18]
```diff
@@ -16,104 +16,121 @@
...
             "name": "MQ_SERIALIZABLE_PACKAGES",
             "value": "",
             "required": false
         },
         {
-            "description": "Size of persistent storage for database volume.",
-            "name": "VOLUME_CAPACITY",
+            "displayName": "A-MQ Volume Size",
+            "description": "Size of the volume used by A-MQ for persisting messages.",
+            "name": "AMQ_VOLUME_CAPACITY",
```
Comment (rcernich) [inline amq/amq62-persistent-ssl.json]: We cannot change the names of the parameters without verifying backward compatibility.  Please revert this back to VOLUME_CAPACITY.

File: decisionserver/decisionserver62-https-s2i.json [hunk 19]
```diff
@@ -34,112 +35,130 @@
...
         },
         {
+            "displayName": "Context Directory",
             "description": "Path within Git project to build; empty for root project directory.",
             "name": "CONTEXT_DIR",
             "value": "decisionserver/hellorules",
             "required": false
         },
         {
+            "displayName": "Queues",
```
Comment (rcernich) [inline decisionserver/decisionserver62-https-s2i.json]: Since we're using A-MQ Queues/Topics for A-MQ templates, do you think it makes sense to use HornetQ Queues/Topics as well?  Lower down, we have HornetQ Password, so...  I think it's fine to just use Queues and Topics for both brokers TBH.

File: decisionserver/decisionserver63-amq-s2i.json [hunk 20]
```diff
@@ -16,167 +16,194 @@
...
         },
         {
+            "displayName": "Context Directory",
             "description": "Path within Git project to build; empty for root project directory.",
             "name": "CONTEXT_DIR",
             "value": "decisionserver/hellorules",
             "required": false
         },
         {
+            "displayName": "JNDI Name",
```
Comment (rcernich) [inline decisionserver/decisionserver63-amq-s2i.json]: For these, I think something like JMS Connection Factory JNDI Name might be more descriptive.

File: processserver/processserver63-amq-mysql-persistent-s2i.json [hunk 21]
```diff
@@ -77,208 +83,243 @@
...
         },
         {
+            "displayName": "Database Volume Capacity",
             "description": "Size of persistent storage for database volume.",
             "name": "VOLUME_CAPACITY",
             "value": "512Mi",
             "required": true
         },
         {
+            "displayName": "JNDI Name",
```
Comment (rcernich) [inline processserver/processserver63-amq-mysql-persistent-s2i.json]: This template highlights the importance of a more descriptive display name.

**Comment by rcernich** [comment]
Looks good, just a few more tweaks.

File: amq/amq62-persistent-ssl.json (hunk 18 above)
Comment (jwendell) [inline amq/amq62-persistent-ssl.json]: Right.

File: decisionserver/decisionserver62-https-s2i.json (hunk 19 above)
Comment (jwendell) [inline decisionserver/decisionserver62-https-s2i.json]: Yeah, let's go with the shorter versions

**Comment by jwendell** [comment]
just pushed for a new round

File: datagrid/datagrid65-mysql-persistent.json (hunk 9 above)
Comment (jwendell) [inline datagrid/datagrid65-mysql-persistent.json]: For this I've created CLOUD-1367
